components/percona-udf/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2023 Percona LLC and/or its affiliates. All rights reserved.
+
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 2 of
+# the License.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+MYSQL_ADD_COMPONENT(percona_udf
+  percona_udf.cc
+  fnv1a_udf.cc
+  fnv_udf.cc
+  murmur_udf.cc
+  MODULE_ONLY)
+
```

Comment:
Also consider making all symbols hidden by default
```
set_target_properties(component_masking_functions PROPERTIES
  CXX_VISIBILITY_PRESET hidden
  VISIBILITY_INLINES_HIDDEN YES
)
```
Probably for backward compatibility (for those who still want to use this .so as a source of user-defined functions via INSTALL FUNCTION), we can also make
* fnv_64
* fnv_64_init
* fnva_64
* fnva_64_init
* murmur_hash
* murmur_hash_init
explicitly exported (`visibility=default`)

Reply:
8.1.0 sets `-fvisibility=hidden` for components by default:
https://github.com/inikep/percona-server/blob/internal-8.1.0-linear/cmake/component.cmake#L115